iothub/device/devdoc/API Design.md

```diff
@@ -14,72 +15,72 @@ public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken =
 /// </summary>
 /// <param name="propertyName">Property name.</param>
 /// <param name="propertyValue">Property value.</param>
+/// <param name="propertyConvention">A convention handler that defines serializer to use for the properties.</param>
 /// <param name="componentName">The component name this property belongs to.</param>
-/// <param name="conventionHandler">A convention handler that defines the content encoding and serializer to use for the properties.</param>
 /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
-public Task UpdatePropertyAsync(string propertyName, object propertyValue, string componentName = default, IConventionHandler conventionHandler = default, CancellationToken cancellationToken = default);
+public Task UpdatePropertyAsync(string propertyName, object propertyValue, PropertyConvention propertyConvention, string componentName = default, CancellationToken cancellationToken = default);
```

Comment:
Q - do we want to accept reported properties to push as key-value or as PropertyCollection type?

Reply:
key-value -> we'll serialize (based on what you tell us) vs PropertyCollection will accept a serialized string as input -> user will have to serialize and send

Reply:
IDictionary<string, object> properties, PropertyConvention propertyConvention,  vs PropoertyCollection - decice internally (managed) since it is essentially serialization detail

Reply:
keep PropertyCollection  only